core/src/main/java/org/bitcoinj/core/AbstractBlockChain.java

```diff
@@ -832,7 +832,7 @@ private void handleNewBestChain(StoredBlock storedPrev, StoredBlock newChainHead
             for (Iterator<StoredBlock> it = newBlocks.descendingIterator(); it.hasNext();) {
                 cursor = it.next();
                 Block cursorBlock = cursor.getHeader();
-                if (expensiveChecks && cursorBlock.time().getEpochSecond() <= getMedianTimestampOfRecentBlocks(cursor.getPrev(blockStore), blockStore))
+                if (expensiveChecks && cursorBlock.time().compareTo(getMedianTimestampOfRecentBlocks(cursor.getPrev(blockStore), blockStore)) <= 0)
```

Comment:
isBefore/isAfter. It's hard for me to even read this code to decide which one it should be 🤣

Reply:
See above.